bundler/lib/bundler/cli/gem.rb

```diff
@@ -415,6 +425,10 @@ def open_editor(editor, file)
       thor.run(%(#{editor} "#{file}"))
     end
 
+    def rust_builder_required_rubygems_version
+      "3.3.11"
```

Comment:
Should we also fail to generate the gem if the user does not have a RubyGems version installed that knows Rust?

If we did this, we could also install the version of Bundler (like we do know) and RubyGems (to make sure it's up to date enough) that the user is running in CI configuration, to make sure things work in CI just like in their local environment.

Reply:
Actually, every RubyGems version that knows Rust also installs a default Bundler version that can auto-switch, so if we do this, I think we can completely avoid the `gem install bundler` step in CI configuration.

Reply:
The CI setup is just example, I would not like to spend a lot of time to make it bullet-proof. Anyway the problem I have faced was related to older Ruby using default RubyGems on CI with no Rust builder support.

IMHO it would be enough to make a comment in the gemspec explaning why this entry is in there.

Reply:
Not sure if my suggestion was clear.

I'm suggesting to add a CI step that runs `gem update --system #{Gem.rubygems_version}`, so that we keep generating CI configuration with deterministic versions (like we did until now), and also with an modern enough RubyGems version that has Rust builder support.

With that we keep the CI configuration using fixed RubyGems versions instead of rolling RubyGems versions, and also guarantee that RubyGems version is modern enough for Rust.

The above would work best if we refused to generate Rust gems that are not going to work anyways because the user does not have an up to date enough RubyGems, because otherwise we would be also generating CI configuration that does not work with Rust either.

I do like to encourage deterministic CI configurations like we've done until now, but I guess it's no big deal if it's necessary to unlock them to make this work without a lot of extra effort.

Reply:
Can we at least keep the CI untouched for the Ruby case, i.e., using fixed RubyGems and Bundler versions? That should be an straightforward change, happy to do it myself!

Reply:
Resolved now! 💪